src/compute/cast/dictionary_to.rs

```diff
@@ -88,8 +89,9 @@ where
             Box::new(values.data_type().clone()),
             is_ordered,
         );
-        // some of the values may not fit in `usize` and thus this needs to be checked
-        DictionaryArray::try_new(data_type, casted_keys, values.clone())
+        // Safety
+        // we errored if any cast overflowed
+        unsafe { DictionaryArray::try_new_unchecked(data_type, casted_keys, values.clone()) }
```

Comment:
The branch above checks the null count. This ensures us that we don't have any overflowing integers, so the invariants of the struct are held and we don't need to check it.

Reply:
P.S. This function is not used. Do we want it for the public API or can we remove it?